homeassistant/components/xbox/__init__.py

```diff
@@ -89,3 +118,83 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
         hass.data[DOMAIN].pop(entry.entry_id)
 
     return unload_ok
+
+
+@dataclass
+class XboxData:
+    """Xbox dataclass for update coordinator."""
+
+    status: SmartglassConsoleStatus
+    app_details: Optional[Product]
+
+
+class XboxUpdateCoordinator(DataUpdateCoordinator):
+    """Store Xbox Console Status."""
+
+    def __init__(
+        self,
+        hass: HomeAssistantType,
+        client: XboxLiveClient,
+        consoles: SmartglassConsoleList,
+    ) -> None:
+        """Initialize."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=timedelta(seconds=10),
+        )
+        self.data: Dict[str, XboxData] = {}
+        self.client: XboxLiveClient = client
+        self.consoles: SmartglassConsoleList = consoles
+
+    async def _async_update_data(self) -> Dict[str, XboxData]:
+        """Fetch the latest console status."""
+        new_data: Dict[str, XboxData] = {}
+        for console in self.consoles.result:
+            current_state: Optional[XboxData] = self.data.get(console.id, None)
```

Comment:
`None` is the default value returned if the key is missing in the dict.

Reply:
Fixed in https://github.com/home-assistant/core/pull/41833